strings/group_anagrams.py

```diff
@@ -0,0 +1,58 @@
+from typing import List
+import doctest
+
+def signature(word: str) -> str:
+  """
+  >>> signature("python")
+  'hnopty'
+  >>> signature("anagram")
+  'aaagmnr'
+  >>> signature("cloud")
+  'cdlou'
+  """
+  return ''.join(sorted(word))
+
+def group_anagrams(words: List[str]) -> dict:
+  """
+  >>> group_anagrams(['cloud', 'python', 'could'])
+  {'cdlou': ['cloud', 'could'], 'hnopty': ['python']}
+  >>> group_anagrams(['strap', 'traps', 'maps'])
+  {'aprst': ['strap', 'traps'], 'amps': ['maps']}
+  >>> group_anagrams(['astro', 'plastic', 'kite'])
+  {'aorst': ['astro'], 'acilpst': ['plastic'], 'eikt': ['kite']}
+  """
+  anagrams = dict()
```

Comment:
How could this code be tighter by using:
```suggestion
  anagrams = defaultdict(list)
```
https://docs.python.org/3/library/collections.html#collections.defaultdict